app/models/foreman_openscap/arf_report.rb

```diff
@@ -189,18 +189,22 @@ def equal?(other)
     end
 
     def destroy_from_proxy
-      if host
-        begin
-          openscap_proxy_api.destroy_report(self, ForemanOpenscap::Helper::find_name_or_uuid_by_host(host))
-        rescue Foreman::Exception => e
-          logger.error "Failed to delete report with id #{id} from proxy, cause: #{e.message}"
-          logger.debug e.backtrace.join("\n\t")
-        end
+      if !host
+        destroy_from_proxy_warning "host"
+      elsif !policy
+        destroy_from_proxy_warning "policy"
+      elsif !openscap_proxy
+        destroy_from_proxy_warning "OpenSCAP proxy"
       else
-        logger.error "Failed to delete report with id #{id} from proxy, no host associated with report"
+        openscap_proxy_api.destroy_report(self, ForemanOpenscap::Helper::find_name_or_uuid_by_host(host))
```

Comment:
we're no longer rescuing from errors on proxy? does that mean we'd render ugly error if proxy returns 500 e.g.? previously, the method in case of proxy error returned true (because of `logger.debug`), should we perhaps keep the rescue in this else branch?

Reply:
ah scratch that, I see the rescue is handled in `#destroy_report` already and it returns false